### PR TITLE
added __future__ absolute_import to files identified by pylint

### PR DESCRIPTION
**Pull request by malmstroem**
Added __future__ absolute_import to files identified by pylint. Tests are passing.

### CODE REVIEW

**Comment by riccardomurri** [comment]
Please rebase against current "master" branch before making PRs -- otherwise commits that might have already been merged show up, making the review process more cumbersome...

I'm sending a PR to your working branch with the rebased commits.

**Comment by riccardomurri** [comment]
Hmmm.... I'm afraid this was a bit too mechanical and cannot be merged in this form.

An instance of a problem:

```diff
modified   install.py
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
 #!/usr/bin/env python
 #
 #  Copyright (C) 2015-2019 University of Zurich.
```

Now the `#!` shebang is no longer the first two characters in the
file, which breaks executability on UNIX platforms.  The shebang `#!`,
if present, *must* stay the first line.

Another example:

```diff
modified   gc3libs/utils.py
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
 #! /usr/bin/env python
 """
 Generic Python programming utility functions.
```

Now, [PEP 236][1] states (in the "Syntax" section) that:

> all future_statments must appear near the top of the module. The
> only lines that can appear before a future_statement are:
>
> * The module docstring (if any).

Actually, the docstring is ignored if does *not* appear before the
`from __future__` line, so also in practice the docstring *must*
appear before `from __future__` lines...

Similarly:

```diff
modified   gc3libs/tests/test_core.py
@@ -1,3 +1,4 @@
+from __future__ import absolute_import
 # test_core.py
 # -*- coding: utf-8 -*-
 #
```

The `coding:` statement must appear within the first two lines of the
file ([PEP 263, section "Defining the encoding"][2]), and the
insertion of `from __future__` displaces it to 3rd line.

A minor point: example code and solutions in `docs/` should probably get a different treatment, since they are explained line by line in the documentation.  I guess it's fine to leave them as they are and come back at a later stage.

[1]: https://www.python.org/dev/peps/pep-0236/#id16
[2]: https://www.python.org/dev/peps/pep-0263/#id8

File: gc3apps/bf.uzh.ch/test/pymods/markovChain/mcInterface.py [hunk 1]
```diff
@@ -92,16 +92,16 @@ def mcInterface(method, uncMean, persistence, varCov, nval, verb = 'DEBUG'):
     if not os.path.isfile(os.path.join(os.getcwd(), 'GHQUAD.DAT')):
       shutil.copy(os.path.join(pathToFile, 'GHQUAD.DAT'), os.getcwd())
     markovInputFile = open('PARMS', 'w')
-    print >> markovInputFile, len(nval)
-    print >> markovInputFile, 1
+    markovInputFile.write(len(nval) + "\n")
+    markovInputFile.write(str(1) + "\n")
     for ele in nval.flat:
-      print >> markovInputFile, ele
+      markovInputFile.write(ele + "\n")
```
Comment (riccardomurri) [inline gc3apps/bf.uzh.ch/test/pymods/markovChain/mcInterface.py]: You should be able to do:

```python
print(whatever, file=markovInputFile)
```

here and in similar cases.

Comment (malmstroem) [reply gc3apps/bf.uzh.ch/test/pymods/markovChain/mcInterface.py]: In my first try, I inserted from __future__ import absolute_import where pylint suggested it, but this approach did not pass the tests. So, should I simply add it after the doc-string or before the first import?

I can certainly rebase - did you send a pull request? I didn't receive it.

Comment (riccardomurri) [reply gc3apps/bf.uzh.ch/test/pymods/markovChain/mcInterface.py]: The `from __future__` line should be after the docstring and before the first import.

Pull request is here: https://github.com/malmstroem/gc3pie/pull/1#event-2232615404

**Comment by malmstroem** [comment]
I *think* I have fixed the print(file=) and the location of the absolute_import. I also rebased my fork. I hope all looks as it should now.

**Pull request merged**